### ISSUE TITLE
MySQL datasource allows to customize connection string

### ISSUE THREAD

**Issue by silencev** [grafana/grafana#16539]
**What would you like to be added**:
In source code [here](https://github.com/grafana/grafana/blob/b9cc476eb7aa57ed6714654e6036c73c9d105919/pkg/services/sqlstore/sqlstore.go#L229) and [here] I found the connection string parameters are hardcoded.

In MySQL Go driver, it supports following parameters
https://github.com/go-sql-driver/mysql#allowcleartextpasswords

**Why is this needed**:
In my use case, I need to add **allowCleartextPasswords=true** in connection string.

It would be convenient for users to has its own parameters.

**Comment by alexanderilyin** [comment]
This is required when using MySQL 8.0 as Data Source.

**Comment by soufianez75** [comment]
Hello I have the same error it's possible to tell me how you have corrected the data source ?

**Comment by marefr** [comment]
@silencev @alexanderilyin are you referring to using MySQL as Grafana config database or querying MySQL using a Grafana data source for plotting data in a dashboard? Or both? Based on referenced code I'm not sure.

**Comment by alexanderilyin** [comment]
In my case I was adding a new MySQL 8 data source for plotting some data on graphs. There was a problem with no default timezone set in MySQL DB and due to some other restrictions, my queries from the panel were failing because of timezone problem. I don't recall how I worked it around but probably found a hacky way. If there was a way to add additional params to the connection string it would be easy to deal with but it's not the option due to how its hardcoded in sources when I checked.
